MastodonSDK/Sources/MastodonCore/Service/API/APIService+Block.swift

```diff
@@ -110,3 +149,11 @@ extension APIService {
     }
     
 }
+
+extension Set<Status> {
+    func deleteAllFeedsForBlockOrMute(in managedObjectContext: NSManagedObjectContext) {
+        map { $0.feeds.union($0.reblogFrom.map { $0.feeds }.flatMap { $0 }) }
+        .flatMap { $0 }
+        .forEach(managedObjectContext.delete)
+    }
```

Comment:
Cool. The home timeline feeds now deleted immediately. 

And I'm sorry my explain not cover all cases. Another timeline kind is notification. 
The Notification has two relationship: 
- `.account`
- `.status`: a related status if notification gives

So there are to paths:
1.  `MastodonUser.statuses` ->  (Status.reblogFrom ->) `Status.notifications`-> `Notification.feeds`
2. `MastodonUser.notifications`

Reply:
Ok got it, I've updated the PR. I have to take a break from this now as my brain is slowly melting :-D Maybe we can try out and test the current implementation as-is and improve with a compound predicate later?

Reply:
It's depends on the size of database. I guess use reverse relationship will be always the best choice.